### ISSUE TITLE
Is there a way to ignore files or directories during jets deploy

### ISSUE THREAD

**Issue by sam0x17** [boltops-tools/jets#561]
We have some sub directories in our main jets app that are quite large and don't need to be deployed. Is there the equivalent of a `.jetsignore` file or something where we can specify directories/files that shouldn't be deployed.

**Comment by tongueroo** [comment]
Sure. Currently, anything in `.gitignore` or `.dockerignore` is not included in the deployment package. 

https://github.com/boltops-tools/jets/blob/master/lib/jets/builders/tidy.rb#L36-L37

So can add the files not wanted in the deployment package to one of those. 👍

**Comment by tongueroo** [comment]
Respecting `.jetsignore` now also. Done in #562

**Comment by sam0x17** [comment]
Thanks a bunch. One issue I am still having is it seems the automatically generated `/vendor` gems directory is including test and development mode gems, and is thus blowing throw the lambda size limit upon deployment. I thought gems got stored in S3?

**Comment by sam0x17** [comment]
Actually I think I know what the problem is. This is my current error:
```
=> rsync -a --links /tmp/jets/arist/stage/code/vendor/gems/ruby/2.7.0/ /tmp/jets/arist/stage/opt/ruby/gems/2.7.0/
=> Replacing compiled gems with AWS Lambda Linux compiled versions: /tmp/jets/arist/stage/opt
Checking projects gems for binary serverlessgems...
Over the Lambda size limit of 250MB
Please reduce the size of your code.
Sizes:
Code: 162.9MB - /tmp/jets/arist/stage/code
Gem Layer: 111.5MB - /tmp/jets/arist/stage/opt
Total Package: 274.4MB
Over limit by: 24.4MB
```
Note that upon manual inspection, the size of my `code` directory is under 4 MB, however if you also include the `vendor` directory that is automatically created, `du -sh` reports the size of my `code` directory as 162.9 MB.

@tongueroo I think your code that calculates that the size of code directory + size of gem layer is too big is not taking into account the symlinks in the `vendor` directory, so the gem layer is effectively being counted twice. In reality jets should be able to deploy this just fine, but it is aborting with an over limit error. Thoughts?

**Comment by sam0x17** [comment]
Update: I am only experiencing this issue in Github Actions / ubuntu-latest. Running locally in ubuntu 20.04, it works just fine. Maybe something about the Github Actions container environment messes up `rsync -a --links`

**Comment by sam0x17** [comment]
Yeah I can now confirm -- for whatever reason the `du` command used by by `jets deploy` (and every combination of `du` command line args I've been able to put together) will include the symlinked `vendor/gems/ruby/2.7.0` directory when calculating the size of the `/tmp/jets/arist/stage/code` directory when inside GitHub Actions (as well as on several of my colleagues' MacOS machines), but will exclude it (correctly) on other systems/environments. I've also manually inspected the created directories in GitHub Actions after failure and I can confirm that the symlink is set up correctly and is identical to the one I have when I run locally. So this is some kind of environment-specific bug with `du`.

I think I will submit a pull request for jets that provides a more portable `compute_size` implementation in pure ruby that explicitly doesn't follow symlinks.
